src/library/dlgtagfetcher.cpp

```diff
@@ -250,45 +308,67 @@ void DlgTagFetcher::fetchTagFinished(
 }
 
 void DlgTagFetcher::slotNetworkResult(
-        int httpError, const QString& app, const QString& message, int code) {
+        int httpError,
+        const QString& app,
+        const QString& message,
+        int code) {
     m_networkResult = httpError == 0 ? NetworkResult::UnknownError : NetworkResult::HttpError;
     m_data.m_pending = false;
-    QString strError = tr("HTTP Status: %1");
-    QString strCode = tr("Code: %1");
-    httpStatus->setText(strError.arg(httpError) + "\n" + strCode.arg(code) + "\n" + message);
-    QString unknownError = tr("Mixxx can't connect to %1 for an unknown reason.");
-    cantConnectMessage->setText(unknownError.arg(app));
-    QString cantConnect = tr("Mixxx can't connect to %1.");
-    cantConnectHttp->setText(cantConnect.arg(app));
+
+    if (m_networkResult == NetworkResult::UnknownError) {
+        QString unknownError = tr("Can't connect to %1 for an unknown reason: %2");
+        loadingProgressBar->setFormat(unknownError.arg(app, message));
+    } else {
+        QString cantConnect = tr("Can't connect to %1: %2");
+        loadingProgressBar->setFormat(cantConnect.arg(app, message));
+    }
+    qWarning() << "Error while getting track metadata!"
```

Comment:
Can you also add m_networkResult to the warning?

Reply:
If the httpError is 0, that means that is unknown right?

Reply:
No, m_networkResult, has no extra information.

Reply:
So on [here](https://github.com/mixxxdj/mixxx/blob/b4dd4743a70ad0b75cfcae7f2ef36b1f2dc011fc/src/library/dlgtagfetcher.cpp#L349), there should be an indicator that we received a network error, right?

Reply:
I have answerd my own comment.
I think we don't need to distinguish different errors. This info is also not relevant for the warning.

Reply:
Yes it is currently used in updateStack()
Maybe you can remove that usage during the refactoring or replace it with a state variable specific for this class.

Reply:
I have tried to refactor the update stack, Before disposing the network result is it okay if I can get a feedback about it? So I can learn if I am on the right way or not.  I will push it as a try-out commit. I would be so happy if I get a feedback on it. If that's the wrong way, I can revert it easily.

What do you think?